internal/proxy/task_search.go

```diff
@@ -276,10 +276,16 @@ func (t *searchTask) PreExecute(ctx context.Context) error {
 
 	if t.request.GetDslType() == commonpb.DslType_BoolExprV1 {
 		annsField, err := funcutil.GetAttrByKeyFromRepeatedKV(AnnsFieldKey, t.request.GetSearchParams())
-		if err != nil {
-			return errors.New(AnnsFieldKey + " not found in search_params")
+		if err != nil || len(annsField) == 0 {
+			if enableMultipleVectorFields {
```

Comment:
this flag only controls if we support multiple vector fields in the schema, seems not very related with search.

Reply:
If we enable multi-vector, an error needs to be reported here to prevent missing changes.